core/jvm/src/main/scala/zio/sql/typetag.scala

```diff
@@ -1,17 +1,20 @@
 package zio.sql
 
+import java.sql.ResultSet
 import java.time._
 import java.util.UUID
-
 import zio.Chunk
 
 trait TypeTagModule {
 
-  type TypeTagExtension[+A]
+  type TypeTagExtension[+A] <: Tag[A]
 
-  sealed trait TypeTag[+A] {
-    private[zio] def cast(a: Any): A = a.asInstanceOf[A]
+  trait Tag[+A] {
+    private[zio] def cast(a: Any): A                                                  = a.asInstanceOf[A]
+    //This is required when implementing database specific custom types.
+    def decode[A, E](column: Either[Int, String], resultSet: ResultSet): Either[E, A] = ???
```

Comment:
this doesn't fit here because all the standard type tags don't need this. If someone calls this on one of the standard types, it goes boom. See if you can find somewhere else for it that won't risk a runtime error.